### PR TITLE
Adding appending functions for urls

### PR DESCRIPTION
**Pull request by danielaRiesgo**
## Summary ##
Adding "url appending" functions for string and numbers (int and uint).

### CODE REVIEW

File: Core/Extensions/Foundation/String.swift [hunk 1]
```diff
@@ -20,6 +20,30 @@ public func / (basePath: String, pathComponent: String) -> String {
...
+     
+     - parameter basePath: base string.
+     - parameter pathComponent: natural number that will be appended.
+     
+     - returns: The new NSURL, with pathComponent appended to baseURL.
+ */
+public func / (basePath: String, pathComponent: UInt) -> String {
+    return (basePath as NSString).appendingPathComponent(String(pathComponent))
+}
+
```
Comment (FranDepascuali) [inline Core/Extensions/Foundation/String.swift]: I think a protocol as a namespace can be used for less repetition.

```swift
public protocol PathAppendable {}

extension Int: PathAppendable {}
extension UInt: PathAppendable {}
extension String: PathAppendable {}

public func / (basePath: String, pathComponent: PathAppendable) -> String {
    return (basePath as NSString).appendingPathComponent(String(describing: pathComponent))
}
```

I think it is okay here to use the `init(describing:)`, as in the docs it says: ```The String(describing:) initializer is the preferred way to convert an instance of any type to a string```

https://developer.apple.com/reference/swift/customstringconvertible

Comment (danielaRiesgo) [inline Core/Extensions/Foundation/String.swift]: I like it.
Although I'd do something like this:
```swift
public protocol PathAppendable {
    func toString() -> String
}

public extension PathAppendable {
    func toString() -> String {
        return String(describing: self)
    }
}
```

 And glad to hear from you !

**Comment by woloxci** [comment]
:+1: There are no linter violations.

File: Core/Extensions/Foundation/String.swift [hunk 2]
```diff
@@ -8,17 +8,7 @@
...
-    
-    - parameter basePath: base string.
-    - parameter pathComponent: string that will be appended.
- 
-    - returns: The new NSURL, with pathComponent appended to baseURL.
- */
-public func / (basePath: String, pathComponent: String) -> String {
-    return (basePath as NSString).appendingPathComponent(pathComponent)
-}
+
```
Comment (damfinkel) [inline Core/Extensions/Foundation/String.swift]: check the amount of whitespaces here

**Pull request merged**